posthog/session_recordings/queries/test/test_session_recording_list_from_query.py

```diff
@@ -4334,5 +4355,204 @@ def test_can_filter_for_flags(self, _name: str, properties: dict, expected: list
             },
         )
 
-        (session_recordings, _, _) = self._filter_recordings_by({"properties": properties})
-        assert sorted([sr["session_id"] for sr in session_recordings]) == expected
+        self._assert_query_matches_session_ids({"properties": properties}, expected)
```

Comment:
then from here on down is new tests to cover the various negated event properties we can query